specs/centos_ci.adoc

```diff
@@ -0,0 +1,135 @@
+// vim: tw=79
+
+= CentOS CI
+
+We need to make sure that new code doesn't break existing codebase and check
+that new features works as early as possible. This can be done by running
+acceptance integration tests automatically, using CentOS CI infrastructure.
+
+== Problem description
+
+CentOS CI
+will be configured to launch acceptance integration test run for every new
+nightly rpm build from tendrl nightly copr project. As set of implemented
+features in tendrl codebase grows, the set of acceptance tests is extended to
+cover new features.
+
+== Use Cases
+
+* Acceptance testing for every new build from tendrl copr nightly project.
+* The same set of tests should be run for stable builds from current tendrl
+  copr as well.
+
+== Proposed change
+
+* New copr tendrl project with nightly builds needs to be created. Right now
+  we have just copr project for
+  https://copr.fedorainfracloud.org/coprs/tendrl/tendrl/[stable tendrl buils],
+  but for the purposes of CI proposed here, we need another one with the latest
+  code.
+
+* When nightly build fails, no acceptance tests are executed and the build
+  needs to be fixed first.
+
+* Acceptance tests executed by CentOS CI system will work with rpm builds only.
+
+* CentOS CI host provisioning system will be utilized for testing machines
+  provisioning.
+
+* Installation and configuration of all Tendrl components should be automated
+  via ansible playbooks/roles from
+  https://github.com/Tendrl/usmqe-setup[usmqe-setup] repository for usage in CI
+  (this contains also installation and configuration of all additional required
+  components including gluster and ceph).
+
+* Tendrl test suite should allow execution of subset of tests marked as
+  acceptance. http://doc.pytest.org/en/latest/example/markers.html[Pytest
+  marker] `@pytest.mark.acceptance` will be used to flag a test case as such.
+
+=== Alternatives
+
+None
+
+=== Data model impact:
+
+None
+
+=== Impacted Modules:
+
+Every module needs to be in good shape from rpmbuild point of view.
```

Comment:
We should in fact automate this as part of the build process itself. `rpmlint` output or some such needs to accompany every build.